packages/babel-helper-define-polyfill-provider/src/utils.ts

```diff
@@ -34,12 +31,16 @@ function resolveId(path) {
   }
 }
 
-export function resolveKey(path: NodePath, computed: boolean = false) {
-  const { node, parent, scope } = path;
-  if (path.isStringLiteral()) return node.value;
-  const { name } = node;
+export function resolveKey(
+  path: NodePath<t.Expression>,
+  computed: boolean = false,
+) {
+  const { scope } = path;
+  if (path.isStringLiteral()) return path.node.value;
   const isIdentifier = path.isIdentifier();
-  if (isIdentifier && !(computed || parent.computed)) return name;
+  if (isIdentifier && !(computed || path.parent.computed)) {
```

Comment:
We can type cast `path.parent` to a `t.MemberExpression`.

Reply:
this one narrowed `path` correctly, but `path.parent` was still just a `Node`. Cast it to a `t.MemberExpression` now